custom-metrics-stackdriver-adapter/README.md

```diff
@@ -0,0 +1,127 @@
+# Custom Metrics - Stackdriver Adapter
+
+Custom Metrics - Stackdriver Adapter is an implementation of [Custom Metrics
+API] based on Stackdriver per-pod metrics. It's purpose is to enable pod
+autoscaling based on Stackdriver custom metrics.
+
+## Usage guide
+
+This guide shows how to set up Custom Metrics - Stackdriver Adapter and export
+metrics to Stackdriver in a compatible way. Once this is done, you can use
+them to scale your application, following [HPA walkthrough].
+
+### Configure cluster
+
+1. Create Kubernetes cluster or use existing one, see [cluster setup].
+   Requirements:
+
+   * Kubernetes version 1.8.1 or newer running on GKE or GCE
+
+   * Monitoring scope `monitoring` set up on cluster nodes. **It is enabled by
+     default, so you should not have to do anything**. See also [OAuth 2.0 API
+     Scopes] to learn more about authentication scopes.
+
+     You can use following commands to verify that the scopes are set correctly:
+     - For GKE cluster `<my_cluster>`, use following command:
+       ```
+       gcloud container clusters describe <my_cluster>
+       ```
+       For each node pool check the section `oauthScopes` - there should be
+       `https://www.googleapis.com/auth/monitoring` scope listed there.
+     - For a GCE instance `<my_instance>` use following command:
+       ```
+       gcloud compute instances describe <my_instance>
+       ```
+       `https://www.googleapis.com/auth/monitoring` should be listed in the
+       `scopes` section.
+
+
+     To configure set scopes manually, you can use:
+     - `--scopes` flag if you are using `gcloud container clusters create`
+       command, see [gcloud
+       documentation](https://cloud.google.com/sdk/gcloud/reference/container/clusters/create).
+     - Environment variable `NODE_SCOPES` if you are using [kube-up.sh script].
+       It is enabled by default.
+     - To set scopes in existing clusters you can use `gcloud beta compute
+       instances set-scopes` command, see [gcloud
+       documentation](https://cloud.google.com/sdk/gcloud/reference/beta/compute/instances/set-scopes).
+
+1. Start *Custom Metrics - Stackdriver Adapter*.
+
+```sh
+kubectl create -f
+https://raw.githubusercontent.com/GoogleCloudPlatform/k8s-stackdriver/master/custom-metrics-stackdriver-adapter/adapter-beta.yaml
+```
+
+### Export metrics to Stackdriver
+
+To learn how to create your custom metric and write your data to Stackdriver,
+follow [Stackdriver custom metrics documentation]. You can also follow
+[Prometheus to Stackdriver documentation] to export metrics exposed by your pods
+in Prometheus format.
+
+The name of your metric must start with custom.googleapis.com/ prefix followed
+by a simple name, as defined in [custom metric naming rules].
+For purpose of this guide, we will use metric `custom.googleapis.com/qps`.
```

Comment:
We say here we will use 'custom.googleapis.com/qps' but we never refer to it later in this guide.

Reply:
Right, this is from the draft document, where I included some examples of HPA configuration. Since we link to HPA documentation instead, I'm removing this.